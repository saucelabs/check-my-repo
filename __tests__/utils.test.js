const { createTempDirectory, printResults, formatedDate } = require('../src/utils')
const fs = require('fs')
jest.mock('fs')

// using TDD to verify if real function is well written
describe('filtering data', () => {
  it('should filter positive results', () => {
    const posResults = Object.values(results)
      .filter(r => r.status == 'passed')
      .map(r => r.rule.name)

    expect(posResults).toEqual(['Rule1', 'Rule5', 'Rule6'])
  })
})

it('should print a result nicely', () => {
  expect(printResults()).toBe(posResults, negResults)
})

it('should create new formated date', () => {
  jest.spyOn(global, 'Date').mockImplementationOnce(() => new Date('2019-05-14T11:01:58.135Z'))
  expect(formatedDate().toEqual(new Date('2019-05-14T11')))
})
