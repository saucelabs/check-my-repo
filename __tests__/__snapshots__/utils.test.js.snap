// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should filter passed results 1`] = `
Array [
  Array [
    "{bgBlue Repository: ,} repository",
  ],
  Array [
    "
        {hex('#FF8800') ðŸš¨ ,} failing-repo",
  ],
  Array [
    "
        {greenBright âœ… ,} passed-repo",
  ],
  Array [
    "
",
  ],
]
`;

exports[`should return global all passed when there is no fail 1`] = `
Array [
  Array [
    "{bgBlue Repository: ,
}
      {greenBright Passed all checks ðŸ¥³ 
} repository",
  ],
]
`;
